[chores] Replace hardcoded colors with CSS variables and format CSS files

atif09 opened the pull request:
## Checklist

- [x] I have read the [OpenWISP Contributing Guidelines](http://openwisp.io/docs/developer/contributing.html).
- [ ] I have manually tested the changes proposed in this pull request.
- [ ] I have written new test cases for new code and/or updated existing tests for changes to existing code.
- [ ] I have updated the documentation.

## Reference to Existing Issue

Closes #1162 .

## Description of Changes
Replaces hardcoded color literals (hex & rgba) in the UI CSS with the project's semantic CSS variables defined in [openwisp-utils#516](https://github.com/openwisp/openwisp-utils/pull/516)

## Screenshot
**N/A**

atif09 committed:
Replace hardcoded colors with CSS variables and format css files

nemesifier reviewed (commented):
@coderabbitai review

piyushdev04 committed:
[fix:tests] Improved robustness of config/tests/test_api.py‎ #442

Problem: Some tests in TestConfigApi assumed a fixed global Template count;
which fail when other modules like openwisp-monitoring add new templates.

Fix: Updated template-related assertions in test_api.py to compare
counts relatively (eg: capture initial_count, assert +1).

Fixes #442

* [tests] Remove unnecessary test settings file #442

The test_settings override is not required to fix the template
counting issue and adds unrelated changes. Removing it to keep
the fix minimal and focused.

Fixes #442

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 committed:
[css] Replace remaining hardcoded colors with CSS variables #1162

Replaced all hardcoded color literals (gray, lightgray, green, yellow, white)
in advanced-mode.css and jsonschema-ui.css with semantic CSS variables from
openwisp-utils for consistent theming support.

Changes:
- advanced-mode.css: Replaced 14 hardcoded colors with appropriate CSS variables
  (--ow-color-fg-medium, --ow-color-fg-light, --ow-color-success,
   --ow-color-warning, --ow-color-white)
- jsonschema-ui.css: Replaced 1 hardcoded white with --ow-color-white

This completes the color variable migration started in the initial commit.

Fixes #1162

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 committed:
[formatting] Replace hardcoded colors with CSS variables #1162

Replaced hardcoded color values with OpenWISP CSS variables
to support theming and maintain consistency with the design system.

Fixes #1162

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 commented on openwisp_controller/config/static/config/css/admin.css:
```diff
@@ -116,26 +116,26 @@ input.readonly {
 .djnjc-overlay .close:hover {
-  background-color: #9e47c1;
+  background-color: var(--ow-color-primary);
 }
```
again, is more of a UX improvement which is unrelated to the issue referenced in the PR, which is to replace hardcoded colors with css variables

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 commented:
will be fixing all the mismatches now

atif09 committed:
Merge branch 'master' into issues/1162-replace-hardcoded-colors-with-ow-vars

atif09 commented:
Ive fixed most of the mismatches, need to take a few screenshots of the UI to compare changes and have a few flags (i.e we might need to introduce new variables as the colors dont exist for some of these). Will do it asap

nemesifier committed:
[chores] Adjustments

nemesifier reviewed (approved):
Thanks @atif09, nice work on this one, left a few adjustments on top in 65c01dcc to address the rules we landed on for the same refactor in https://github.com/openwisp/openwisp-firmware-upgrader/pull/389. Quick rundown:

- Stopped using `--ow-btn-danger-bg` for non-button fills/text. Replaced it with `--ow-color-danger` on error text/borders, `--delete-button-bg` / `--delete-button-hover-bg` for the `input.deletelink` button (exact value match for Django admin), and `--message-error-bg` for the import-preview row backgrounds (which already had matching siblings `--message-success-bg` / `--message-warning-bg`).
- Restored a few colors that didn't have a good semantic var via small ad-hoc variables defined at the top of the file: the `.djnjc-preformatted` success/error text on the dark preview overlay, the purple close-X, the slate "preview configuration" button, and the JSON editor's blue accent (border + menu bar). Same pattern for the JSON editor's inactive search highlight and the null syntax color.
- Number and boolean JSON values now share a `--jsoneditor-syntax-value` var mapped to `--ow-color-primary`, so they stay consistent and easy to retheme.
- Picked up a few colors that the original pass missed: `whois.css`, plus the inline colors in `commands.js` (success/error row flash) and `widget.js` (disabled labels and tab border).

After this re-grep across the module's CSS/JS/templates comes back clean.

nemesifier commented:
@coderabbit resolve
